File: SlimFileManager/src/main/java/com/slim/util/MediaStoreUtils.java
package com.slim.util;

import android.content.Context;
import android.database.Cursor;
import android.net.Uri;
import android.provider.OpenableColumns;

public class MediaStoreUtils {

    public static String getFileNameFromUri(Context context, Uri uri) {
        Cursor cursor = context.getContentResolver().query(uri, null, null, null, null);
        if (cursor == null) return "";
        try {
            int nameIndex = cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME);
            cursor.moveToFirst();
            return cursor.getString(nameIndex);
        } finally {
            cursor.close();
        }
    }
}

File: manager/src/main/java/com/slim/util/MediaStoreUtils.kt
package com.slim.util

import android.content.Context
import android.database.Cursor
import android.net.Uri
import android.provider.OpenableColumns

object MediaStoreUtils {

    fun getFileNameFromUri(context: Context, uri: Uri): String {
        val cursor = context.contentResolver.query(uri, null, null, null, null) ?: return ""
        try {
            val nameIndex = cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME)
            cursor.moveToFirst()
            return cursor.getString(nameIndex)
        } finally {
            cursor.close()
        }
    }
}
